.bazelrc

```
@@ -6,106 +6,106 @@
 # factored the stamping done by this to avoid excessive build performance impact
 # and so enable stamping with it by default. CI and systems especially dependent
 # on caching should explicitly use `--nostamp`.
-build --workspace_status_command=./scripts/workspace_status.py
-build --stamp
+common --workspace_status_command=./scripts/workspace_status.py
+common --stamp
 
 # Provide aliases for configuring the release and pre-release version being
 # built. For documentation of these flags, see //bazel/version/BUILD.
-build --flag_alias=release=//bazel/version:release
-build --flag_alias=pre_release=//bazel/version:pre_release
-build --flag_alias=rc_number=//bazel/version:rc_number
-build --flag_alias=nightly_date=//bazel/version:nightly_date
+common --flag_alias=release=//bazel/version:release
+common --flag_alias=pre_release=//bazel/version:pre_release
+common --flag_alias=rc_number=//bazel/version:rc_number
+common --flag_alias=nightly_date=//bazel/version:nightly_date
 
 # Support running clang-tidy with:
 #   bazel build --config=clang-tidy -k //...
 # See: https://github.com/erenon/bazel_clang_tidy
-build:clang-tidy --aspects @bazel_clang_tidy//clang_tidy:clang_tidy.bzl%clang_tidy_aspect
-build:clang-tidy --output_groups=report
-build:clang-tidy --@bazel_clang_tidy//:clang_tidy_config=//:clang_tidy_config
-build:clang-tidy --action_env=PATH --host_action_env=PATH
+common:clang-tidy --aspects @bazel_clang_tidy//clang_tidy:clang_tidy.bzl%clang_tidy_aspect
+common:clang-tidy --output_groups=report
+common:clang-tidy --@bazel_clang_tidy//:clang_tidy_config=//:clang_tidy_config
+common:clang-tidy --action_env=PATH --host_action_env=PATH
 
 # This warning seems to incorrectly fire in this build configuration, despite
 # not firing in our normal builds.
-build:clang-tidy --copt=-Wno-unknown-pragmas
+common:clang-tidy --copt=-Wno-unknown-pragmas
 
 # --config=non-fatal-checks makes CHECK failures not terminate compilation.
-build:non-fatal-checks --per_file_copt=common/check_internal.cpp@-DCARBON_NON_FATAL_CHECKS
+common:non-fatal-checks --per_file_copt=common/check_internal.cpp@-DCARBON_NON_FATAL_CHECKS
 
 # Provide an alias for controlling the `carbon_*` Bazel rules' configuration. We
 # enable use of the target config here to make our build and tests more
 # efficient, see the documentation in //bazel/carbon_rules/BUILD for details.
-build --flag_alias=use_target_config_carbon_rules=//bazel/carbon_rules:use_target_config_carbon_rules
+common --flag_alias=use_target_config_carbon_rules=//bazel/carbon_rules:use_target_config_carbon_rules
 build --use_target_config_carbon_rules
```

```suggestion
common --use_target_config_carbon_rules
```

Can this be switched?

---

No, it fails because bazel doesn't know which commands will support this flag, and it tries to use it in places where the --flag_alias was not added. Specifically I believe in query during precommit.

---

Thanks for the clarification; before merging, can you add a comment explaining, since this now stands out from the rest of the file?

---

Yep done